src/curies/discovery.py

```diff
@@ -0,0 +1,93 @@
+"""Discovery new entries for a Converter."""
+
+from collections import defaultdict
+from typing import TYPE_CHECKING, Any, Iterable, Mapping, Sequence, Set
+
+from curies import Converter, Record
+
+if TYPE_CHECKING:
+    import rdflib
+
+__all__ = [
+    "discover",
+    "discover_from_rdf",
+]
+
+
+def discover_from_rdf(converter: Converter, graph: "rdflib.Graph", **kwargs: Any) -> Converter:
+    """Discover new URI prefixes from an RDFLib triple store."""
+    uris = set(_yield_uris(graph))
+    return discover(converter, uris, **kwargs)
+
+
+def _yield_uris(graph: "rdflib.Graph") -> Iterable[str]:
+    import rdflib
+
+    for parts in graph.triples((None, None, None)):
+        for part in parts:
+            if isinstance(part, rdflib.URIRef):
+                yield str(part)
+
+
+def discover(
```

Comment:
I think "discover" is not really the best word to express what is happening here. I think `generate_missing_prefixes` would be a clearer name. But I am not too hung up on this here, your design.

Reply:
I want the mindset for this functionality to be that it's a data exploration tool. It can be used to understand the URIs you have, and then push you forwards to make a decision on how you want to build up an (extended) prefix map.

I don't want `curies` to be the place where people look for hacky solutions to their semantics issues.

Reply:
More docs added to better contextualize this name and give an example of how a data scientist should use this workflow